File: python/codeeval/080_uri_comparison.py
import urllib
import sys

def uc(test):
    s1, s2 = test.split(';')
    return (urllib.parse.unqoute(s1)).lower() == (urllib.parse.unqoute(s2)).lower()

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(uc(test))

File: ruby/codeeval/080_uri_comparison.rb
require 'uri'
T = 'True'
F = 'False'
def uc(line)
    s1, s2 = line.strip().split(';')
    return URI.decode(s1).downcase.gsub(':80/', '/') == URI.decode(s2).downcase.gsub(':80/', '/') ? T : F
end

File.open(ARGV[0]).each_line do |line|
    puts(uc(line))
end
